spotbugs/src/main/java/edu/umd/cs/findbugs/util/SecurityManagerHandler.java

```diff
@@ -0,0 +1,60 @@
+package edu.umd.cs.findbugs.util;
+
+/**
+ * Since Java 17, the security manager is deprecated for removal and invoking related methods
+ * causes a warning to be printed to the console. This intermediate disables the use of
+ * security manager-related APIs on Java 17 or later, unless using the security manager is
+ * explicitly configured by setting the <i>edu.umd.cs.findbugs.securityManagerDisabled</i>
+ * property.
+ */
+public class SecurityManagerHandler {
+
+    /**
+     * Determines if the security manager is used by SpotBugs.
+     */
+    public static boolean SECURITY_MANAGER_DISABLED;
+
+    static {
+        boolean securityManagerDisabled;
+        try {
+            String property = System.getProperty("edu.umd.cs.findbugs.securityManagerDisabled");
+            if (property != null) {
+                securityManagerDisabled = Boolean.parseBoolean(property);
+            } else {
+                String version = System.getProperty("java.version");
+                if (version.startsWith("1.")) {
+                    version = version.substring(2, 3);
+                } else {
+                    int index = version.indexOf(".");
+                    if (index != -1) {
+                        version = version.substring(0, index);
+                    }
+                }
+                securityManagerDisabled = Integer.parseInt(version) > 16;
```

Comment:
nitpick: [GraalVM native image does not support SecurityManager](https://www.graalvm.org/22.0/security-guide/#security-manager-and-untrusted-code) even in v11 release, so it would be better to check `java.vm.name` or something to judge.

Reply:
I changed to use the utility. As for Graal, I added a check to only set the security manager to null if one is currently set. This works for Graal and some other platforms like Android, as they still provide compatible API where the security manager is always `null`.

Reply:
Please push the change to GitHub 😀